test/shared/src/main/scala/zio/test/TestAspect.scala

```diff
@@ -223,6 +222,46 @@ object TestAspect extends TimeoutVariants {
    */
   val flaky: TestAspectPoly = eventually
 
+  /**
+   * An aspect that runs a test on a forked fiber. This is handy to use with aspects that repeat tests such as
```

Comment:
I would say something about how this ensures that each test is run with a fresh set of `FiberRefs`. Normally when people think of `fork` they expect it to introduce additional parallelism but that is not the case here. Then you can mention `fiberTime` as one example of this.